r_markdown/ef_summary.Rmd

```diff
@@ -0,0 +1,82 @@
+---
+title: "Emission Factor Summary"
+output:
+  html_document:
+    toc: true
+    toc_float: true
+    theme: united
+---
+
+```{r, echo=FALSE, warning=FALSE, message=FALSE, results='markup', cache=FALSE}
+  library(tidyverse)
+  library(forcats)
+```
+
+```{r, echo=FALSE, warning=FALSE, message=FALSE, results='markup', cache=FALSE}
+  source("../r_scripts/R_functions.R")
+```
+
+## Load and tidy data
+
+```{r, echo=FALSE, warning=FALSE, message=FALSE, results='markup', cache=FALSE}
+  load("../r_files/emissions.Rda")  # load emissions data
+  load("../r_files/samples.Rda")  # load sample meta data
+  load("../r_files/fuel_burnt.Rda")  # fuel use meta data
+```
+
+* Output emissions column names 
+
+```{r, echo=FALSE, warning=FALSE, message=FALSE, results='markup', cache=FALSE}
+  colnames(emissions)
+```
+
+* Filter out bad tests
+
+```{r, echo=TRUE, warning=FALSE, message=FALSE, results='markup', cache=FALSE}
+  emissions <- dplyr::filter(emissions, qc != "bad")
+```
+
+* Merge with fuel burnt data
+
+```{r, echo=TRUE, warning=FALSE, message=FALSE, results='markup', cache=FALSE}
+  emissions <- dplyr::left_join(emissions, dplyr::select(fuel_burnt, id,
+                                                 mass_fuel, lhv,
+                                                 carbon_fraction), by = "id") %>%
+           dplyr::left_join(dplyr::select(samples, id, stove,
+                                          fuel, fuelcat, stovecat,
+                                          stove_fuel), by = "id")
+```
+
+## Calculate emissions factors
+ 
+```{r, echo=TRUE, warning=FALSE, message=FALSE, results='markup', cache=FALSE}
+  emissions <- dplyr::mutate(emissions, mass_ef = 
+                           (mass_emitted * 1e-3) / (mass_fuel * 1e-3)) %>%  # mg / kg
+           dplyr::mutate(energy_ef = (mass_ef * 1e-3)*(lhv * 1e-3))  # g / MJ
+```
+
+* emissions are given in units of miligrams per kilogram of fuel burned and grams per MJ delivered by the fuel
+
+# CO data
+
+* Plot emissions factors by mass, energy of fuel and energy delivered
+
+```{r, echo=FALSE, warning=FALSE, message=FALSE, results='markup', cache=FALSE, fig.width = 14,  fig.height = 10}
+  plot_mass_ef(dplyr::filter(emissions, grepl('^co2|^co|^ch4', pol)), 'PM', "mass")
```

Comment:
we should deal with the "NA" upstream of this Rm, try and find where it appears...